Tools/ci-run.sh

```diff
@@ -3,125 +3,164 @@
 GCC_VERSION=${GCC_VERSION:=8}
 
 # Set up compilers
-if [ "${OS_NAME##ubuntu*}" == "" -a "$TEST_CODE_STYLE" != "1" ]; then
+if [[ $TEST_CODE_STYLE == "1" ]]; then
+  echo "Skipping compiler setup"
+elif [[ $OSTYPE == "linux-gnu"* ]]; then
+  echo "Setting up linux compiler"
   echo "Installing requirements [apt]"
   sudo apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
   sudo apt update -y -q
   sudo apt install -y -q ccache gdb python-dbg python3-dbg gcc-$GCC_VERSION || exit 1
-  if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $BACKEND == *"cpp"* ]]; then
```

Comment:
Are the quotes needed here? They look funny between the wildcards.

Reply:
Yes, it's a string matching. Well, the syntax is pretty .. interesting i would say but .. that's how it goes..